app/models/submission_completion.rb

```diff
@@ -1,12 +1,16 @@
 class SubmissionCompletion
-  attr_reader :submission
+  attr_reader :submission, :user
 
-  def initialize(submission)
+  def initialize(submission, user)
     @submission = submission
+    @user = user
   end
 
   def perform!
     submission.reviewed_and_accepted! do
+      submission.submitted_by = user
+      submission.submitted_at = Time.zone.now
+      submission.save
```

Comment:
I wonder, should this be a `save!` instead so that an exception is raised if the save fails for whatever reason? Otherwise we may end up transitioning submissions to completed without recording the submitter details.

Reply:
Ah yes, so looks like the way that aasm works differently to how I imagined – according to the [README](https://github.com/aasm/aasm/blob/master/README.md), if you pass a block to a transition the block is executed after the transition succeeds. I think you're right, we should wrap the whole thing in our own transaction and not bother with the block on the transition call.

Reply:
Just noting that we've made the decision to proceed and merge this PR and address refactoring `SubmissionCompletion#perform!` in a separate PR.